src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -262,6 +262,56 @@ FabricInfo * RetrieveCurrentFabric(CommandHandler * aCommandHandler)
     return Server::GetInstance().GetFabricTable().FindFabricWithIndex(index);
 }
 
+void FailSafeCleanup(FailSafeContext & failSafeContext)
+{
+    FabricInfo * fabricInfo = Server::GetInstance().GetFabricTable().FindFabricWithIndex(failSafeContext.GetFabricIndex());
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Call to FailSafeCleanup");
+
+    if (fabricInfo)
+    {
+        // If an AddNOC or UpdateNOC command has been successfully invoked, terminate all CASE sessions associated with the Fabric
+        // whose Fabric Index is recorded in the Fail-Safe context (see ArmFailSafe Command) by clearing any associated Secure
+        // Session Context at the Server.
+        if (failSafeContext.NocCommandHasBeenInvoked())
+        {
+            CASESessionManager * caseSessionManager = Server::GetInstance().GetCASESessionManager();
+
+            if (caseSessionManager)
+            {
+                caseSessionManager->ReleaseSession(fabricInfo->GetPeerId());
```

Comment:
This is only one session. It's all the fabric's session. Suggest `ReleaseSessions` method for a given `FabricIndex` maybe?